l5kit/l5kit/rasterization/semantic_rasterizer.py

```diff
@@ -80,51 +62,157 @@ def __init__(
         return sem_im.astype(np.float32) / 255
 
+    def render_semantic_map(
+        self, center_world: np.ndarray, world_to_image_space: np.ndarray, tl_faces: np.ndarray
+    ) -> np.ndarray:
+        """Renders the semantic map at given x,y coordinates.
+
+        Args:
+            center_world (np.ndarray): XY of the image center in world ref system
+            world_to_image_space (np.ndarray):
+        Returns:
+            np.ndarray: RGB raster
+
+        """
+
+        img = 255 * np.ones(shape=(self.raster_size[1], self.raster_size[0], 3), dtype=np.uint8)
+
+        # filter using half a radius from the center
+        raster_radius = float(np.linalg.norm(self.raster_size * self.pixel_size)) / 2
+
+        # get active traffic light faces
+        active_tl_ids = set(filter_tl_faces_by_status(tl_faces, "ACTIVE")["gid"].tolist())
+
+        # plot lanes
+        lanes_lines = defaultdict(list)
+
+        for idx in elements_within_bounds(center_world, self.bounds_info["lanes"]["bounds"], raster_radius):
+            lane = self.proto_API[self.bounds_info["lanes"]["ids"][idx]].element.lane
+
+            # get image coords
+            lane_coords = self.proto_API.get_lane_coords(self.bounds_info["lanes"]["ids"][idx])
+            xy_left = cv2_subpixel(transform_points(lane_coords["xyz_left"][:, :2], world_to_image_space))
+            xy_right = cv2_subpixel(transform_points(lane_coords["xyz_right"][:, :2], world_to_image_space))
+            lanes_area = np.vstack((xy_left, np.flip(xy_right, 0)))  # start->end left then end->start right
+
+            # Note(lberg): this called on all polygons skips some of them, don't know why
+            cv2.fillPoly(img, [lanes_area], (17, 17, 31), lineType=cv2.LINE_AA, shift=CV2_SHIFT)
+
+            lane_type = "default"  # no traffic light face is controlling this lane
+            # TODO should we change the next lane (lanes_ahead) status?
+            lane_tl_ids = set([MapAPI.id_as_str(la_tc) for la_tc in lane.traffic_controls])
+            for tl_id in lane_tl_ids.intersection(active_tl_ids):
+                if self.proto_API.is_traffic_face_red(tl_id):
+                    lane_type = "red"
+                elif self.proto_API.is_traffic_face_green(tl_id):
+                    lane_type = "green"
+                elif self.proto_API.is_traffic_face_yellow(tl_id):
+                    lane_type = "yellow"
+
+            lanes_lines[lane_type].extend([xy_left, xy_right])
+
+        cv2.polylines(img, lanes_lines["default"], False, (255, 217, 82), lineType=cv2.LINE_AA, shift=CV2_SHIFT)
+        cv2.polylines(img, lanes_lines["green"], False, (0, 255, 0), lineType=cv2.LINE_AA, shift=CV2_SHIFT)
+        cv2.polylines(img, lanes_lines["yellow"], False, (255, 255, 0), lineType=cv2.LINE_AA, shift=CV2_SHIFT)
+        cv2.polylines(img, lanes_lines["red"], False, (255, 0, 0), lineType=cv2.LINE_AA, shift=CV2_SHIFT)
```

Comment:
Can you please define the colour tuples as constants with names like RED, GREEN. It will make this a lot more readable.

Reply:
ditto :(